chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -96,29 +94,24 @@ def forward_gpu(self, inputs):
         log_y = softmax_log(x, self.use_cudnn)
         self.y = cupy.exp(log_y)
         if getattr(self, 'normalize', True):
-            count = float((t != self.ignore_label).sum())
+            count = (t != self.ignore_label).sum()
         else:
-            count = t.shape[0]
-        self.count = count
-
-        if count == 0:
-            return cupy.zeros((), dtype=x.dtype),
```

Comment:
I think we should treat `count == 0` as a special case for readability, if we can ignore the difference of computing complexity between the creation of zero-filled array and reduction kernel.

The reason for it is the interpretation of `count`. In usual case, `count` is the number of labels that are not ignored. But if all labels are ignored and `self.normalize` is `True`, `count` is 1, although I think we can expect is `0`. So by treating the latter case as a special one, we make the interpretation of `count` more complex.